### ISSUE TITLE
Can't generate tree-sitter-wasm: failed to rename wasm output file

### ISSUE THREAD

**Issue by TijlE-1951707** [tree-sitter/tree-sitter#3195]
### Problem

I failed at building the wasm for python. Instead I tried downloading them [from this repository here](https://github.com/tree-sitter/tree-sitter.github.io/raw/master/tree-sitter-python.wasm) in the hopes that it would work, but it doens't. I get the following errors:

<details>
<summary>
Uncaught runtime errors:
</summary>

```
ERROR
need to see wasm magic number
n@http://localhost:3000/static/js/bundle.js:95097:28
getDylinkMetadata@http://localhost:3000/static/js/bundle.js:95104:16
loadWebAssemblyModule@http://localhost:3000/static/js/bundle.js:95319:45
./node_modules/web-tree-sitter/tree-sitter.js/init/initPromise</load/TreeSitter<@http://localhost:3000/static/js/bundle.js:96856:55
ERROR
Aborted(CompileError: wasm validation error: at offset 4: failed to match magic number). Build with -sASSERTIONS for more info.
abort@http://localhost:3000/static/js/bundle.js:94975:164
./node_modules/web-tree-sitter/tree-sitter.js/init/initPromise</instantiateArrayBuffer/TreeSitter<@http://localhost:3000/static/js/bundle.js:95012:67
```
</details>

When I try building the file, I run into this annoying, recognizable OCI problem from docker.

```console
$ npx tree-sitter build --wasm node_modules/tree-sitter-python
Error: error stat'ing file `/path-to-my-app/node_modules/tree-sitter-python/src`: Permission denied: OCI permission denied
emcc command failed
```

I have very little knowledge of docker, and often run into a Permission denied: OCI permission denied problem in my `/media/user/Data` partition, where my project is. Usually, I solve this by mounting the directory in another point (`/Data1`) with full permissions and run the commands from there, something like the following:

```console
$ sudo mount --bind /path-to-my-app/ /Data1
$ cd /Data1
$ npx tree-sitter build --wasm node_modules/tree-sitter-python
failed to rename wasm output file

Caused by:
    No such file or directory (os error 2)
```

I'm hoping that "failed to rename wasm output file" indicates that the file is to be found somewhere, and I only have to manually extract it. I'd like to know where I can find it, if so. I'm trying to prototype something using ReactJs + Typescript, so I don't mind it being not production ready.

### Steps to reproduce

I was following the steps in [Web Tree-sitter](https://github.com/tree-sitter/tree-sitter/blob/master/lib/binding_web/README.md).
```
npm install web-tree-sitter
npx tree-sitter build --wasm node_modules/tree-sitter-python # javascript behaves the same
```

### Expected behavior

The `tree-sitter-python.wasm` file to be generated.

### Tree-sitter version (tree-sitter --version)

├── tree-sitter-cli@0.22.2
├── tree-sitter-javascript@0.20.4
├── tree-sitter-python@0.21.0
├── web-tree-sitter@0.20.8

### Operating system/version

Zorin OS 17.1 (Linux 6.5)

**Comment by sogaiu** [comment]
Regarding:

> $ npx tree-sitter build --wasm node_modules/tree-sitter-python

Does it work any better if you also specify `-o tree-sitter-python.wasm`?

**Comment by TijlE-1951707** [comment]
It does! I removed the original file and copied to generated file to the destination.

I tried the same code when replaced by my own generated one and I get the same runtime errors displayed in the React project's gui.

```
Uncaught runtime errors:
ERROR
need to see wasm magic number
n@http://localhost:3000/static/js/bundle.js:95098:28
getDylinkMetadata@http://localhost:3000/static/js/bundle.js:95105:16
loadWebAssemblyModule@http://localhost:3000/static/js/bundle.js:95320:45
./node_modules/web-tree-sitter/tree-sitter.js/init/initPromise</load/TreeSitter<@http://localhost:3000/static/js/bundle.js:96857:55
ERROR
Aborted(CompileError: wasm validation error: at offset 4: failed to match magic number). Build with -sASSERTIONS for more info.
abort@http://localhost:3000/static/js/bundle.js:94976:164
./node_modules/web-tree-sitter/tree-sitter.js/init/initPromise</instantiateArrayBuffer/TreeSitter<@http://localhost:3000/static/js/bundle.js:95013:67
```

This is the code that triggers the error ([based on the README's setup example code for npm](https://github.com/tree-sitter/tree-sitter/blob/master/lib/binding_web/README.md#setup)).
```
import Parser from "web-tree-sitter";

Parser.init().then(() => { /* the library is ready */ });
const Python = await Parser.Language.load('./lib/tree-sitter/tree-sitter-javascript.wasm'); // file has been removed
```

Just to test if it would be cached somewhere, I deleted the file and the same runtime errors occur. I don't know what to run to uncache it, if that's the case. (I ran `npm run build` and `npm start`, and I closed the browser. I believe the browser I'm using to test it, Librewolf, doesn't have persistent cookies.)

**Comment by TijlE-1951707** [comment]
Not sure if this closed because I was misinterpreted. I had removed `tree-sitter` from the following packages, since it wasn't necessary to use `web-tree-sitter`.
├── tree-sitter-cli@0.22.2
├── tree-sitter-javascript@0.20.4
├── tree-sitter-python@0.21.0
├── web-tree-sitter@0.20.8
The error `need to see wasm magic number` hasn't disappeared. However, in meanwhile, I actually ended up using something server-side, so I don't mind now. Also, I'll be installing another distribution eventually, because Zorin OS 17's _upgrade_ came with a few issues, so in the future I wouldn't be able to help at all.

Thanks for your time, nonetheless!
